pkg/analysis/passes/osvscanner/osvscanner.go

```diff
@@ -0,0 +1,193 @@
+package osvscanner
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+)
+
+var (
+	missingOSVScanner                  = &analysis.Rule{Name: "osvscanner-missing-binary", Severity: analysis.Warning}
+	scanningFailure                    = &analysis.Rule{Name: "osvscanner-failed", Severity: analysis.Warning}
+	scanningParseFailure               = &analysis.Rule{Name: "osvscanner-parse-failed", Severity: analysis.Warning}
+	scanningSucceeded                  = &analysis.Rule{Name: "osvscanner-succeeded", Severity: analysis.Warning}
+	osvScannerCriticalSeverityDetected = &analysis.Rule{Name: "osvscanner-critical-severity-vulnerabilities-detected", Severity: analysis.Error}
+	osvScannerHighSeverityDetected     = &analysis.Rule{Name: "osvscanner-high-severity-vulnerabilities-detected", Severity: analysis.Warning}
+	osvScannerModerateSeverityDetected = &analysis.Rule{Name: "osvscanner-moderate-severity-vulnerabilities-detected", Severity: analysis.Warning}
+	osvScannerLowSeverityDetected      = &analysis.Rule{Name: "osvscanner-low-severity-vulnerabilities-detected", Severity: analysis.Warning}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "osv-scanner",
+	Requires: []*analysis.Analyzer{archive.Analyzer},
+	Run:      run,
+	Rules: []*analysis.Rule{
+		missingOSVScanner,
+		osvScannerCriticalSeverityDetected,
+		osvScannerHighSeverityDetected,
+		osvScannerModerateSeverityDetected,
+		osvScannerLowSeverityDetected,
+		scanningFailure,
+		scanningParseFailure,
+		scanningSucceeded},
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveDir := pass.ResultOf[archive.Analyzer].(string)
+	// we're detecting only go.mod and package.lock (this can be changed to use defaults)
+	lockFile := filepath.Join(archiveDir, "go.mod")
+	if _, err := os.Stat(lockFile); err != nil {
+		// check for yarn.lock
+		lockFile = filepath.Join(archiveDir, "yarn.lock")
+		if _, err := os.Stat(lockFile); err != nil {
```

Comment:
should this check return here since there's nothing the validator can do if there's no go.mod or yarn.lock to check?

Reply:
that's what it is doing for the reportall - if that isn't enabled yes it should be returning here